src/app/calendar/parcelCalendarFunctions.ts

```diff
@@ -1,23 +1,37 @@
 import { Schema } from "@/database_utils";
 import supabase from "@/supabaseClient";
 import { CalendarEvent } from "@/components/Calendar/Calendar";
+import { RainbowPalette } from "styled-components";
```

Comment:
Why is this coming from styled-components?

Reply:
the colorArray content should all from keyof RainbowPalette (I can't just randomly put a color like "darkPink" because it is not in the RainbowPalette and it will cause error when I try to use `chosenTheme.rainbow.darkPink,background` for instance

Reply:
Yeah but I don't get how this isn't imported from somewhere else? Fair enough if it's right, just I've never seen anything but styled come from styled components

Reply:
It is defined in styled.d.ts so it is now "defined" in styled-component
This change was done in the theme ticket so it can allow type checking in styled-components (e.g. if I type rainbow.darkPink.background it will show me error in the IDE before runtime)